Refactor project structure and add SectionController

iuliaszarics opened the pull request:
- Removed folder references for "Controllers" and "Repositories" in Duo.Api.csproj.
- Updated namespace in CourseCompletion.cs from Duo.Models to Duo.Api.Models.
- Introduced SectionController with CRUD endpoints for managing sections.
- Enhanced code documentation with XML comments for better readability.

iuliaszarics committed:
Refactor project structure and add SectionController

- Removed folder references for "Controllers" and "Repositories" in Duo.Api.csproj.
- Updated namespace in CourseCompletion.cs from Duo.Models to Duo.Api.Models.
- Introduced SectionController with CRUD endpoints for managing sections.
- Enhanced code documentation with XML comments for better readability.

UjfalusiAbel commented on Duo.Api/Controllers/SectionController.cs:
```diff
@@ -0,0 +1,104 @@
+    /// <param name="section">Section data from form.</param>
+    /// <returns>Created section.</returns>
+    [HttpPost("add")]
+    public async Task<IActionResult> Add([FromForm] Section section)
```
You should have as parameters all the fields that are needed in a ```Section```. You should also use the repository in order to add a new section, not directly add in controller

UjfalusiAbel commented on Duo.Api/Controllers/SectionController.cs:
```diff
@@ -0,0 +1,104 @@
+    /// </summary>
+    /// <param name="id">ID of the section.</param>
+    [HttpGet("get")]
+    public async Task<IActionResult> Get([FromQuery] int id)
```
You should use the repository in order to get a new section, not directly do it in controller. 
```id``` should be [FromForm] as well

UjfalusiAbel commented on Duo.Api/Controllers/SectionController.cs:
```diff
@@ -0,0 +1,104 @@
+    /// </summary>
+    /// <returns>List of all sections.</returns>
+    [HttpGet("list")]
+    public async Task<IActionResult> List()
```
You should use repository in order to access data

UjfalusiAbel commented on Duo.Api/Controllers/SectionController.cs:
```diff
@@ -0,0 +1,104 @@
+    /// <param name="id">Id of the section to be deleted</param>
+    /// <returns>Status of the deletion.</returns>
+    [HttpDelete("delete")]
+    public async Task<IActionResult> Delete([FromQuery] int id)
```
You should use repository for deletion of data

UjfalusiAbel commented on Duo.Api/Controllers/SectionController.cs:
```diff
@@ -0,0 +1,104 @@
+    /// <param name="section">Updated section data from form</param>
+    /// <returns>Status of the update</returns>
+    [HttpPut("update")]
+    public async Task<IActionResult> Update([FromForm] Section section)
```
You should use repository for updating data

UjfalusiAbel reviewed (changes requested):
Please rewrite using the ```Repository``` class and ```IRepository``` interface. You should not access ```DataContext``` directly in controller

iuliaszarics committed:
Update project references and namespace usage

- Added `Swashbuckle.AspNetCore` package reference (v6.6.2) in `Duo.Api.csproj`.
- Removed folder structure for `Controllers` and `Repositories`.
- Changed namespace reference in `CourseCompletion.cs` from `Duo.Models` to `Duo.Api.Models`.
